### PR TITLE
Add test 'Sort carriers by' in FO

### PR DESCRIPTION
**Pull request by nesrineabdmouleh**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | Add test 'Sort carriers by' in FO
| Type?             | refacto
| Category?         | TE
| BC breaks?        | no
| Deprecations?     | no
| Fixed ticket?     | no
| How to test?      | no
| Possible impacts? | TEST_PATH="functional/BO/09_shipping/02_preferences/02_carrierOptions/02_sortBy.js" URL_FO=yourShopURL npm run specific-test

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/22659)

### CODE REVIEW

**Review by matthieu-rolland — DISMISSED** [review]

File: tests/UI/campaigns/functional/BO/09_shipping/02_preferences/02_carrierOptions/02_sortBy.js [hunk 1]
```diff
@@ -0,0 +1,256 @@
+require('module-alias/register');
```
Comment (boubkerbribri) [inline tests/UI/campaigns/functional/BO/09_shipping/02_preferences/02_carrierOptions/02_sortBy.js]: I would suggest to maybe change the file name and all descriptions to `updateCarriersSortOption`

File: tests/UI/campaigns/functional/BO/09_shipping/02_preferences/02_carrierOptions/02_sortBy.js [hunk 2]
```diff
@@ -0,0 +1,256 @@
...
+
+    numberOfCarriers = await carriersPage.resetAndGetNumberOfLines(page);
+    await expect(numberOfCarriers).to.be.above(0);
+  });
+
+  const carriersNames = [
+    Carriers.cheapCarrier.name,
+    Carriers.lightCarrier.name,
+  ];
+  carriersNames.forEach((carrierName, index) => {
```
Comment (boubkerbribri) [inline tests/UI/campaigns/functional/BO/09_shipping/02_preferences/02_carrierOptions/02_sortBy.js]:
```suggestion
  [
    Carriers.cheapCarrier.name,
    Carriers.lightCarrier.name,
  ].forEach((carrierName, index) => {
```

File: tests/UI/campaigns/functional/BO/09_shipping/02_preferences/02_carrierOptions/02_sortBy.js [hunk 3]
```diff
@@ -0,0 +1,256 @@
...
+      await testContext.addContextItem(this, 'testIdentifier', `enableCarrier${index}`, baseContext);
+
+      const isActionPerformed = await carriersPage.updateEnabledValue(page, 1, true);
+
+      if (isActionPerformed) {
+        const resultMessage = await carriersPage.getTextContent(
+          page,
+          carriersPage.alertSuccessBlock,
+        );
+        await expect(resultMessage).to.contains(carriersPage.successfulUpdateStatusMessage);
```
Comment (boubkerbribri) [inline tests/UI/campaigns/functional/BO/09_shipping/02_preferences/02_carrierOptions/02_sortBy.js]:
```suggestion
        const resultMessage = await carriersPage.getTextContent(
          page,
          carriersPage.alertSuccessBlock,
        );
        
        await expect(resultMessage).to.contains(carriersPage.successfulUpdateStatusMessage);
```

File: tests/UI/campaigns/functional/BO/09_shipping/02_preferences/02_carrierOptions/02_sortBy.js [hunk 4]
```diff
@@ -0,0 +1,256 @@
...
+      await testContext.addContextItem(this, 'testIdentifier', `disableCarrier${index}`, baseContext);
+
+      const isActionPerformed = await carriersPage.updateEnabledValue(page, 1, false);
+
+      if (isActionPerformed) {
+        const resultMessage = await carriersPage.getTextContent(
+          page,
+          carriersPage.alertSuccessBlock,
+        );
+        await expect(resultMessage).to.contains(carriersPage.successfulUpdateStatusMessage);
```
Comment (boubkerbribri) [inline tests/UI/campaigns/functional/BO/09_shipping/02_preferences/02_carrierOptions/02_sortBy.js]:
```suggestion
        const resultMessage = await carriersPage.getTextContent(
          page,
          carriersPage.alertSuccessBlock,
        );
        
        await expect(resultMessage).to.contains(carriersPage.successfulUpdateStatusMessage);
```

**Review by boubkerbribri — COMMENTED** [review]
Few comments.
And Please rebase and recheck your PR because I believe the names of these functions have changed : `updateEnabledValue`, `getToggleColumnValue`

File: tests/UI/campaigns/functional/BO/09_shipping/02_preferences/02_carrierOptions/02_sortBy.js [hunk 5]
```diff
@@ -0,0 +1,256 @@
...
+const foCheckoutPage = require('@pages/FO/checkout');
+
+// Import data
+const {Carriers} = require('@data/demo/carriers');
+const {DefaultAccount} = require('@data/demo/customer');
+
+// Import test context
+const testContext = require('@utils/testContext');
+
+const baseContext = 'functional_BO_shipping_preferences_carriersOptions_sortBy';
```
Comment (Progi1984) [inline tests/UI/campaigns/functional/BO/09_shipping/02_preferences/02_carrierOptions/02_sortBy.js]:
```suggestion
const baseContext = 'functional_BO_shipping_preferences_carrierOptions_sortBy';
```

**Review by Progi1984 — COMMENTED** [review]
Feedbacks from @boubkerbribri

File: tests/UI/campaigns/functional/BO/09_shipping/02_preferences/02_carrierOptions/02_sortBy.js (hunk 2 above)
Comment (nesrineabdmouleh) [reply tests/UI/campaigns/functional/BO/09_shipping/02_preferences/02_carrierOptions/02_sortBy.js]: I used the const `carrierNames` 2 times in the test( when I enable and disable carriers)
But I can delete `sortCarriersBy`-> done

File: tests/UI/campaigns/functional/BO/09_shipping/02_preferences/02_carrierOptions/02_sortBy.js (hunk 4 above)
Comment (nesrineabdmouleh) [reply tests/UI/campaigns/functional/BO/09_shipping/02_preferences/02_carrierOptions/02_sortBy.js]: Done thx :+1:

File: tests/UI/campaigns/functional/BO/09_shipping/02_preferences/02_carrierOptions/02_sortBy.js (hunk 5 above)
Comment (nesrineabdmouleh) [reply tests/UI/campaigns/functional/BO/09_shipping/02_preferences/02_carrierOptions/02_sortBy.js]: Done thx :+1:

File: tests/UI/campaigns/functional/BO/09_shipping/02_preferences/02_carrierOptions/02_sortBy.js (hunk 1 above)
Comment (nesrineabdmouleh) [reply tests/UI/campaigns/functional/BO/09_shipping/02_preferences/02_carrierOptions/02_sortBy.js]: Done :+1:

**Review by Progi1984 — DISMISSED** [review]

**Review by NeOMakinG — APPROVED** [review]

**Review by Progi1984 — APPROVED** [review]

**Review by boubkerbribri — APPROVED** [review]

**Comment by Progi1984** [comment]
Thanks @nesrineabdmouleh

**Pull request merged**